examples/tree_2d_dgsem/elixir_kpp.jl

```diff
@@ -0,0 +1,152 @@
+using Trixi
+using OrdinaryDiffEq
+using Plots
+using LinearAlgebra
+
+###############################################################################
+# Definition of the 2D scalar "KPP equation"
+#
+# See: Kurganov, A., Petrova, G., and Popov, B. (2007).
+# Adaptive Semidiscrete Central-Upwind Schemes for Nonconvex Hyperbolic Conservation Laws,
+# SIAM Journal on Scientific Computing, 29(6), 2381--2401
+
+struct KPPEquation2D <: Trixi.AbstractEquations{2, 1} end
+
+# The KPP flux is F(u) = (sin(u), cos(u))
+@inline function Trixi.flux(u, orientation::Integer, ::KPPEquation2D)
+    if orientation == 1
+        return SVector(sin(u[1]))
+    else
+        return SVector(cos(u[1]))
+    end
+end
+
+# Since the KPP problem is a scalar equation, the entropy-conservative flux is uniquely determined
+@inline function Trixi.flux_ec(u_ll, u_rr, orientation::Integer, ::KPPEquation2D)
+    if abs(u_ll[1] - u_rr[1]) < 1e-12
+        return 0.5*(flux(u_ll, orientation, KPPEquation2D())+flux(u_rr, orientation, KPPEquation2D()))
+    else
+        factor = 1.0/(u_rr[1] - u_ll[1])
+        if orientation == 1
+            return SVector(factor*(-cos(u_rr[1]) + cos(u_ll[1])))
+        else
+            return SVector(factor*(sin(u_rr[1]) - sin(u_ll[1])))
+        end
+    end
+end
+
+# Wavespeeds
+const kpp_wavespeed = 1.0
+@inline Trixi.max_abs_speeds(u, ::KPPEquation2D) = (kpp_wavespeed, kpp_wavespeed)
+@inline Trixi.max_abs_speed_naive(u_ll, u_rr, orientation::Integer, ::KPPEquation2D) = kpp_wavespeed
+@inline Trixi.max_abs_speed_naive(u_ll, u_rr, normal_direction::AbstractVector, ::KPPEquation2D) = kpp_wavespeed * norm(normal_direction)
+
+# Compute entropy: we use the square entropy
+@inline Trixi.entropy(u::Real, ::KPPEquation2D) = 0.5 * u^2
+@inline Trixi.entropy(u, ::KPPEquation2D) = entropy(u[1], equation)
+
+# Convert between conservative, primitive, and entropy variables. The conserved quantity "u" is also
+# considered the "primitive variable". Since we use the square entropy, "u" is also the entropy
+# variable.
+@inline Trixi.cons2prim(u, ::KPPEquation2D) = u
+@inline Trixi.cons2entropy(u, ::KPPEquation2D) = u
+
+# Total energy
+@inline Trixi.energy_total(u::Real, ::KPPEquation2D) = 0.5 * u^2
+@inline Trixi.energy_total(u, equation::KPPEquation2D) = energy_total(u[1], equation)
```

Comment:
Out of curiosity: Is this needed in this elixir?

Reply:
No, it's not needed, I deleted it now.